reactor-netty-http/src/test/java/reactor/netty/http/Http2Tests.java

```diff
@@ -414,4 +415,34 @@ public void testHttp2ForMemoryLeaks() throws Exception {
 		}
 		System.gc();
 	}
+
+	@Test
+	public void testHttpClientDefaultSslProvider() {
+		HttpClient client = HttpClient.create()
+		                              .wiretap(true);
+
+		doTestHttpClientDefaultSslProvider(client.protocol(HttpProtocol.H2));
+		doTestHttpClientDefaultSslProvider(client.protocol(HttpProtocol.H2)
+		                               .secure());
+		doTestHttpClientDefaultSslProvider(client.secure()
```

Comment:
What's the difference with the line above?

Reply:
- When the configuration is `client.protocol(HttpProtocol.H2)` and the url starts with `https` then we add the default `SslProvider` in `HttpClientConnect` - https://github.com/reactor/reactor-netty/pull/1273/files#diff-23ed16da152b0327324bd951bd35198dR209
- When the configuration is `client.protocol(HttpProtocol.H2).secure()` or `client.secure().protocol(HttpProtocol.H2)`, we ensure that the corresponding methods take into account the current configuration. See `HttpClient#protocol` and `HttpClient#secure`.
Let's say we have `client.secure().protocol(HttpProtocol.H2)`, when `secure` is invoked we do not have info for the protocol so we add the default `SslProvider` for `HTTP/1.1`, later when the protocol is set we need to change the `SslProvider`